pkg/controller/rotation/handle_secrets_rotate.go

```diff
@@ -0,0 +1,417 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rotation
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/google/exposure-notifications-server/pkg/base64util"
+	"github.com/google/exposure-notifications-server/pkg/logging"
+	"github.com/google/exposure-notifications-verification-server/internal/project"
+	"github.com/google/exposure-notifications-verification-server/pkg/database"
+	"github.com/hashicorp/go-multierror"
+	"go.opencensus.io/stats"
+)
+
+// HandleRotateSecrets handles secrets rotation.
+func (c *Controller) HandleRotateSecrets() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		logger := logging.FromContext(ctx).Named("rotation.HandleRotateSecrets")
+		logger.Debugw("starting")
+		defer logger.Debugw("finishing")
+
+		ok, err := c.db.TryLock(ctx, secretsRotationLock, c.config.MinTTL)
+		if err != nil {
+			logger.Errorw("failed to acquire lock", "error", err)
+			c.h.RenderJSON(w, http.StatusInternalServerError, err)
+			return
+		}
+		if !ok {
+			logger.Debugw("skipping (too early)")
+			c.h.RenderJSON(w, http.StatusOK, fmt.Errorf("too early"))
+			return
+		}
+
+		// If there are any errors, return them
+		if err := c.RotateSecrets(ctx); err != nil {
+			logger.Errorw("failed to rotate secrets", "error", err)
+			c.h.RenderJSON(w, http.StatusInternalServerError, err)
+			return
+		}
+
+		stats.Record(ctx, mSecretsSuccess.M(1))
+		c.h.RenderJSON(w, http.StatusOK, nil)
+	})
+}
+
+// RotateSecrets triggers a secret rotation. It does not take out a lock nor
+// does it return an HTTP response. This is primarily used so other functions
+// can perform initials ecrets bootstrapping.
+func (c *Controller) RotateSecrets(ctx context.Context) error {
```

Comment:
I wonder if you want to name this something scarier like 

`BootstrapRotateSecrets` `UnlockedRotateSecrets` `RecklessRotateSecrets`

Reply:
It still won't rotate the secrets if they haven't met the TTL